agent/engine/docker_task_engine_test.go

```diff
@@ -3174,8 +3198,43 @@ func TestMonitorExecAgentRunning(t *testing.T) {
 				Return(execcmd.NotRestarted, nil).
 				Times(1)
 		}
-		taskEngine.(*DockerTaskEngine).monitorExecAgentRunning(ctx, testTask, testTask.Containers[0])
-		// TODO: [ecs-exec] Add validation for calls to Control Plane if agent was restarted
+
+		waitDone := make(chan struct{})
+
+		expectedManagedAgent := apicontainer.ManagedAgent{
+			Status: apicontainerstatus.ManagedAgentRunning,
+		}
+		if tc.expectContainerEvent {
+			go func(stateChangeEvents <-chan statechange.Event, expectedManagedAgent apicontainer.ManagedAgent) {
+				for event := range stateChangeEvents {
+					t.Logf("event: %v", event)
+					if containerEvent, ok := event.(api.ContainerStateChange); ok {
+						if containerEvent.ManagedAgents != nil {
+							// there is currently only ever a single managed agent
+							assert.Equal(t, expectedManagedAgent.Status, containerEvent.ManagedAgents[0].Status)
+							t.Log("asserted")
```

Comment:
leftover from debugging?

Reply:
sorry I pushed my changes too early -- they weren't ready for review yet.